src/models/transfers.js

```diff
@@ -318,6 +318,13 @@ function * fulfillTransfer (transferId, fulfillmentUri) {
   const fulfillment = ConditionFulfillment.fromDataExternal(fulfillmentUri)
   fulfillment.setTransferId(transferId)
   const existingFulfillment = yield db.transaction(function * (transaction) {
+    // Set isolation level to avoid reading "prepared" transaction that is currently being
+    // executed by another request. This ensures the transfer can be fulfilled only once.
+    if (db.type === 'oracle' || db.type === 'postgres') {
```

Comment:
Is this right? The client for postgres is "pg" and the client for oracle is "strong-oracle". Also, can you add the helper function as discussed earlier?